framework/db/QueryBuilder.php

```diff
@@ -260,6 +260,11 @@ public function build($query, $params = [])
             $sql = "($sql){$this->separator}$union";
         }
 
+        $with = $this->buildWithQueries($query->withQueries, $params);
+        if ($with !== '') {
+            $sql = "$with{$this->separator}$sql";
```

Comment:
```suggestion
            $sql = $with . $this->separator . $sql;
```

Reply:
I wrote it in a same way as union concatenation above

Reply:
I've benchmark both constructions: "$with{$this->separator}$sql" - is near 15% faster for same memory consumption

Reply:
Oh my :smile: 
Ok, leave your version